src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,24 +1,174 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 /**
  * <p>Реалізувати свою HashMap, а саме методи `put(K key, V value)`, `getValue()` та `getSize()`.
  * Дотриматися основних вимог щодо реалізації мапи (initial capacity, load factor, resize...)
  * За бажанням можна реалізувати інші методи інтрефейсу Map.</p>
  */
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int DEFAULT_BUCKETS = 16;
+
+    private Node[] table;
+    private int buckets;
+    private int capacity;
+    private int size;
+
+    public MyHashMap(int buckets) {
+        this.buckets = buckets;
+        this.capacity = 0;
+        this.size = 0;
+        this.table = new Node[buckets];
+    }
+
+    public MyHashMap() {
+        this.buckets = DEFAULT_BUCKETS;
+        this.capacity = 0;
+        this.size = 0;
+        this.table = new Node[DEFAULT_BUCKETS];
+    }
+
+    private int getHash(K key) {
+        int k = Math.abs(key.hashCode());
+        k ^= (k >>> 20) ^ (k >>> 12);
+        return (key == null) ? 0 : k ^ (k >>> 7) ^ (k >>> 4);
+    }
+
+    private int getIndex(int hash, int tableLength) {
+        int i = hash % tableLength;
+        return i;
+    }
+
+    private void checkCapacity() {
+        if (capacity >= buckets * LOAD_FACTOR) {
+            grow();
+        }
+    }
+
+    private boolean checkBucketValue(int index) {
+        return table[index] != null;
+    }
+
+    private void grow() {
+        size = 0;
+        capacity = 0;
+        buckets *= 2;
+        Node[] oldTable = table;
+        table = new Node[buckets];
+        for (Node<K, V> node : oldTable) {
+            while (node != null) {
+                put(node.key, node.value);
+                node = node.nextNode;
+            }
+        }
+    }
+
+    private void putForNull(V value) {
```

Comment:
do we need this method? let's rewrite method `getHash` and remove method `putForNull` and `getForNull`

Reply:
done)